File: Backend-MERN/services/aiService.js
// Works with AND without array, returns fields your frontend needs
export const analyzeWithAI = async (input) => {
  const logsText = Array.isArray(input)
   ? input.map(l => l.message || l.level || JSON.stringify(l)).join(" ")
    : (typeof input === 'string'? input : input?.message || "");

  const low = logsText.toLowerCase();
  const total = Array.isArray(input)? input.length : 1;

  let severity = "LOW";
  let rootCause = "Routine operation";
  let fix = "No action needed. Continue monitoring.";
  let confidence = "89%";

  if (low.includes("critical") || low.includes("crash")) {
    severity = "CRITICAL";
    rootCause = "Service crash / DB disconnection";
    fix = "Immediate: Restart Node.js server (pm2 restart). Check MongoDB Atlas IP whitelist and DB credentials in.env. Verify connection pool size and restart service.";
    confidence = "96%";
  } else if (low.includes("error") || low.includes("erkor") || low.includes("db failed") || low.includes("fail")) {
    severity = "CRITICAL";
    rootCause = "DB Connection Lost / Query Failed";
    fix = "Restart DB and check connection pool. Check logs for connection timeout. Increase poolSize in mongoose config and add retryWrites=true. Verify Atlas Network Access.";
    confidence = "92%";
  } else if (low.includes("warn")) {
    severity = "MEDIUM";
    rootCause = "Resource limit / High memory warning";
    fix = "Clear old logs from Atlas, add pagination on frontend, monitor CPU usage. Add index on timestamp field for faster queries.";
    confidence = "84%";
  } else {
    severity = "LOW";
    rootCause = "Normal user activity / Server started";
    fix = "No fix required. System operating normally. Keep monitoring analytics dashboard.";
    confidence = "78%";
  }

  // Return ALL possible field names so any frontend version works
  return {
    rootCause,
    fix,
    suggestion: fix,
    suggestedFix: fix,
    severity,
    level: severity,
    confidence,
    totalAnalyzed: total,
    summary: rootCause,
    aiPowered: true,
    model: "LogGuard Rule-AI v1"
  };
};

export const analyzeLogWithAI = analyzeWithAI;
export const analyzeLogsAI = analyzeWithAI;
export default analyzeWithAI;

File: Backend-Java/src/main/java/com/logguard/service/AiService.java
package com.logguard.service;

import org.springframework.stereotype.Service;
import java.util.HashMap;
import java.util.Map;

@Service
public class AiService {
    
    private final AlertService alertService;

    public AiService(AlertService alertService) {
        this.alertService = alertService;
    }

    public Map<String, String> analyze(String logMessage) {
        String low = logMessage.toLowerCase();
        String severity = "LOW";
        String rootCause = "Routine operation";
        String fix = "No action needed";

        if (low.contains("erkor")) {
            low = low.replace("erkor", "error");
        }

        if (low.contains("critical") || low.contains("crash") || low.contains("down")) {
            severity = "CRITICAL";
            rootCause = "Service crash / DB disconnection";
            fix = "Restart service and check MongoDB connection";
        } else if (low.contains("error") || low.contains("fail")) {
            severity = "CRITICAL";
            rootCause = "DB Connection Lost";
            fix = "Restart DB and check connection pool";
        } else if (low.contains("warn")) {
            severity = "MEDIUM";
            rootCause = "High resource usage";
            fix = "Scale up or optime query";
        }

        if (severity.equals("CRITICAL")) {
            System.out.println("🚨 AiService detected CRITICAL: " + logMessage);
            alertService.checkAndAlert(severity, logMessage);
        }

        Map<String, String> result = new HashMap<>();
        result.put("severity", severity);
        result.put("rootCause", rootCause);
        result.put("fix", fix);
        result.put("confidence", "92%");
        return result;
    }
}